src/zenml/integrations/huggingface/materializers/huggingface_datasets_materializer.py

```diff
@@ -43,12 +44,12 @@ def handle_input(self, data_type: Type[Any]) -> Dataset:
             The dataset read from the specified dir.
         """
         super().handle_input(data_type)
-        temp_dir = TemporaryDirectory()
+        temp_dir = mkdtemp()
         io_utils.copy_dir(
             os.path.join(self.artifact.uri, DEFAULT_DATASET_DIR),
-            temp_dir.name,
+            temp_dir,
         )
-        return load_from_disk(temp_dir.name)
+        return load_from_disk(temp_dir)
```

Comment:
This works, but it is a little unclean as we don't cleanup this directory after it's not needed anymore (= the user step code has finished). Not necessary for this PR, but I'll create a ticket for this as it's something that we also do in other materializers

Reply:
@schustmi how can we do that while making sure it still works when a step is cached? because the other function used before to create temp dir was able to clean up but this fails when the user uses a cached step.

Reply:
Not within this function, but we could have a `cleanup()` method that get's called on all materializers after a step is finished. https://zenml.atlassian.net/browse/ENG-1538